File: frontend/src/types/document.ts
export type DocumentCategory =
  | 'IDENTITY'
  | 'ADDRESS'
  | 'POLICY'
  | 'CLAIM'
  | 'MEDICAL'
  | 'VEHICLE'
  | 'FINANCIAL'
  | 'OTHER';

export type VerificationStatus = 'PENDING' | 'UNDER_REVIEW' | 'VERIFIED' | 'REJECTED';

export interface DocumentVersion {
  id: string;
  version: number;
  fileName: string;
  fileSize: number;
  uploadedAt: string;
  status: VerificationStatus;
}

export interface DocumentMetadata {
  id: string;
  title: string;
  category: DocumentCategory;
  mimeType: string;
  uploadedBy: string;
  uploadedAt: string;
  status: VerificationStatus;
  verifiedBy?: string;
  verifiedAt?: string;
  remarks?: string;
  currentVersion: number;
  versions: DocumentVersion[];
}

export interface DocumentTimelineEvent {
  id: string;
  type: 'UPLOADED' | 'SUBMITTED' | 'ASSIGNED' | 'REVIEWED' | 'VERIFIED' | 'REJECTED';
  description: string;
  timestamp: string;
  actor: string;
}

File: backend/app/schemas/document.py
from app.schemas.base import BaseSchema, AuditSchema
from typing import List, Optional
from uuid import UUID

class DocumentVersionRead(AuditSchema):
    version: int
    file_name: str
    file_size: int

class DocumentRead(AuditSchema):
    title: str
    category: str
    mime_type: str
    status: str
    file_size: int
    current_version: int
    verified_by: Optional[UUID] = None
    verified_at: Optional[str] = None
    remarks: Optional[str] = None
    versions: List[DocumentVersionRead] = []
